File: ApiSuvesaPos/Negocio/Logica/Empresa.vb
Namespace Logica
    Public Class Empresa

        Private db As Datos.Class.Empresa

        Sub New()
            Me.db = New Datos.Class.Empresa
        End Sub

        Public Function Buscar(porNombre As Boolean, filtro As String) As List(Of Datos.Models.Empresa)
            Return Me.db.Buscar(porNombre, filtro)
        End Function

        Public Function Crear(empresa As Datos.Models.Empresa) As String
            Return Me.db.Crear(empresa)
        End Function

        Public Function Editar(id As Long, empresa As Datos.Models.Empresa) As String
            Dim res As String = Me.db.Editar(id, empresa)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

        Public Function Eliminar(id As Long) As String
            Dim res As String = Me.db.Borrar(id)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

    End Class
End Namespace

File: ApiSuvesaPos/Datos/Class/Empresa.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.Models;
using Microsoft.EntityFrameworkCore;

namespace Datos.Class
{
  public  class Empresa
    {

        private SeePOSContext entities;

        public Empresa()
        {
            entities = new SeePOSContext();
        }
        
        public int Crear(Models.Empresa empresa) // crear carta apartir de objeto de creacion
        {
            try
            {
                entities.Empresas.Add(empresa);
                return entities.SaveChanges();

            }
            catch (Exception ex)
            {
                throw ex;
            }

        }

        public List<Models.Empresa> Buscar(bool porNombre, string filtro) //obtener lista CartasdeExoneracion 
        {
            try
            {
                List<Models.Empresa> result;

                if (porNombre == true)
                {
                    //busca por la descripcion de la marca
                    var temp = from c in entities.Empresas
                                   // usar un like
                               select c;

                    result = temp.ToList<Models.Empresa>();
                }
                else
                {
                    //busca por el id de la tabla
                    var temp = from c in entities.Empresas
                               where c.Id == Convert.ToInt32(filtro) //convertir el datos de string a long
                               select c;
                    result = temp.ToList<Models.Empresa>();
                }


                if (result.Count > 0)
                {
                    return result;
                }
                else
                {
                    return result = null;
                }


            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public int Editar(long id, Models.Empresa empresa)
        {
            try
            {
                var p = entities.Empresas.Find(id);
                Models.Empresa viejo = p;

                if (viejo != null)
                {
                    viejo.Empresa1 = empresa.Empresa1;
                    viejo.Actual = empresa.Actual;
                    viejo.Cedula = empresa.Cedula;                

                    entities.Entry(viejo).State = EntityState.Modified;

                    return entities.SaveChanges();
                }
                else
                {
                    return 0;// no se encotro el registro solicitado.
                }

            }
            catch (Exception ex)
            {

                throw ex;
            }
        }

        public int Borrar(long id) // borrar Actividad apartir del id Actividad


        {
            try
            {
                var p = entities.Empresas.Find(id);
                entities.Remove(p);
                return entities.SaveChanges();

            }
            catch (Exception ex)
            {

                throw ex;
            }

        }


    }
}
